proxy_ring_buffer.go

```diff
@@ -0,0 +1,278 @@
+// uses the Ringbuffer for the first time. The most reliable way to do that is to write a dummy item into the Ringbuffer and
+// use the returned sequence as initial  id. On the reading side, this dummy item should be discarded. Please keep in mind that
+// this id is not the sequence of the item you are about to publish but from a previously published item. So it can't be used
+// to find that item.
+//
+// Add returns the sequence number of the added item. You can read the added item using this number.
+func (rb *Ringbuffer) Add(ctx context.Context, item interface{}, overflowPolicy OverflowPolicy) (sequence int64, err error) {
+	elementData, err := rb.validateAndSerialize(item)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	request := codec.EncodeRingbufferAddRequest(rb.name, int32(overflowPolicy), elementData)
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferAddResponse(response), nil
+}
+
+// AddAll
+// Adds all the items of a collection to the tail of the Ringbuffer. A addAll is likely to outperform multiple calls
+// to add(Object) due to better io utilization and a reduced number of executed operations. If the batch is empty,
+// the call is ignored. When the collection is not empty, the content is copied into a different data-structure.
+// This means that: after this call completes, the collection can be re-used. the collection doesn't need to be serializable.
+// If the collection is larger than the capacity of the Ringbuffer, then the items that were written first will be
+// overwritten. Therefor this call will not block. The items are inserted in the order of the Iterator of the collection.
+// If an addAll is executed concurrently with an add or addAll, no guarantee is given that items are contiguous.
+// The result of the future contains the sequenceId of the last written item.
+func (rb *Ringbuffer) AddAll(ctx context.Context, overflowPolicy OverflowPolicy, items ...interface{}) (int64, error) {
+	elementData, err := rb.validateAndSerializeValues(items)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	request := codec.EncodeRingbufferAddAllRequest(rb.name, elementData, int32(overflowPolicy))
+	response, err := rb.invokeOnPartition(ctx, request, rb.partitionID)
+	if err != nil {
+		return ReadResultSetSequenceUnavailable, err
+	}
+	return codec.DecodeRingbufferAddAllResponse(response), nil
+}
+
+// ReadOne
+// Reads one item from the Ringbuffer. If the sequence is one beyond the current tail, this call blocks until an
+// item is added. This method is not destructive unlike e.g. a queue.take. So the same item can be read by multiple
+// readers, or it can be read multiple times by the same reader. Currently, it isn't possible to control how long this
+// call is going to block. In the future we could add e.g. tryReadOne(long sequence, long timeout, TimeUnit unit).
```

Comment:
`tryReadOne(long sequence, long timeout, TimeUnit unit)` part is Java specific. Let's remove the sentence or change the signature to be go compatible.

Reply:
Done.
1ae142f8e57787f00df65b6c9e4888d75ec7b7a6